unitTests/ui/DynamicHierarchicalFacetSearchValueRendererTest.ts

```diff
@@ -0,0 +1,265 @@
+import { $$ } from '../../src/utils/Dom';
+
+function setProperties<T>(obj: T, properties: Partial<T>) {
+  Object.keys(properties).forEach(key => (obj[key] = properties[key]));
+  return obj;
+}
+
+function expectClass(element: HTMLElement, className: string) {
+  expect(element.classList).toContain(className);
+}
+
+function expectChildren(element: HTMLElement, classNames: string[]) {
+  expect(element.childNodes.length).toEqual(classNames.length);
+  return classNames.map((className, index) => {
+    const childAtIndex = element.childNodes.item(index) as HTMLElement;
+    expectClass(childAtIndex, className);
+    return childAtIndex;
+  });
+}
+
+export function DynamicHierarchicalFacetSearchValueRendererTest() {
+  describe('DynamicHierarchicalFacetSearchValueRenderer', () => {
+    let facetValue: DynamicHierarchicalFacetSearchValue;
+    let facet: IDynamicHierarchicalFacet;
+    let renderer: DynamicHierarchicalFacetSearchValueRenderer;
+
+    function initializeFacet(): IDynamicHierarchicalFacet {
+      facet = mockComponent<DynamicHierarchicalFacet>(DynamicHierarchicalFacet);
+      facet.options = {
+        field: '@title',
+        clearLabel: 'begone'
+      };
+      return facet;
+    }
+
+    function initializeFacetValue(parents: string[] = []): DynamicHierarchicalFacetSearchValue {
+      return setProperties((facetValue = mock<DynamicHierarchicalFacetSearchValue>(DynamicHierarchicalFacetSearchValue)), {
+        displayValue: 'A B C',
+        numberOfResults: 1337,
+        fullPath: [...parents, 'abc']
+      });
+    }
+
+    function initializeRenderer(parents: string[] = []) {
+      return (renderer = new DynamicHierarchicalFacetSearchValueRenderer(initializeFacetValue(parents), initializeFacet()));
+    }
+
+    describe('without any parent', () => {
+      beforeEach(() => {
+        initializeRenderer();
+      });
+
+      describe('when selected', () => {
+        beforeEach(() => {
+          renderer.selectAction();
+        });
+
+        it('calls facet.selectPath with the full path', () => {
+          expect(facet.selectPath).toHaveBeenCalledTimes(1);
+          expect(facet.selectPath).toHaveBeenCalledWith(facetValue.fullPath);
+        });
+
+        it('calls facet.enableFreezeFacetOrderFlag', () => {
+          expect(facet.enableFreezeFacetOrderFlag).toHaveBeenCalledTimes(1);
+        });
+
+        it('calls facet.enablePreventAutoSelectionFlag', () => {
+          expect(facet.enablePreventAutoSelectionFlag).toHaveBeenCalledTimes(1);
+        });
+
+        it('calls facet.scrollToTop', () => {
+          expect(facet.scrollToTop).toHaveBeenCalledTimes(1);
+        });
+
+        it('calls facet.triggerNewQuery with a function that calls facetValue.logSelectActionToAnalytics', () => {
+          expect(facet.triggerNewQuery).toHaveBeenCalledTimes(1);
+          const [logAnalytics] = <[Function]>(facet.triggerNewQuery as jasmine.Spy).calls.mostRecent().args;
```

Comment:
You can do that kind of args checking with something similar to : 

```
expect(facet.triggerNewQuery).toHaveBeenCalledWith(...params you expect)
//or
expect(facetValue.logSelectActionToAnalytics).toHaveBeenCalledWith(...)
```

Reply:
The problem is that I'm testing the function that was passed as a parameter.